core/src/main/scala/org/apache/spark/deploy/Client.scala

```diff
@@ -115,27 +121,50 @@ private class ClientActor(driverArgs: ClientArguments, conf: SparkConf)
           System.exit(-1)
         }
         System.exit(0)
-    }
+      }
   }
 
   override def receiveWithLogging: PartialFunction[Any, Unit] = {
 
     case SubmitDriverResponse(success, driverId, message) =>
       println(message)
-      if (success) pollAndReportStatus(driverId.get) else System.exit(-1)
+      if (success) {
+        activeMasterActor = context.actorSelection(sender.path)
+        pollAndReportStatus(driverId.get)
+      } else if (!message.contains("Can only")) {
```

Comment:
Ah, I see, it's from this message: `Can only accept driver submissions in ALIVE state`.